development/workflow.rst

```diff
@@ -0,0 +1,305 @@
+
+We encourage you to enable `Two-Factor Authentication (2FA) for GitHub <https://help.github.com/articles/about-two-factor-authentication/>`_ through your `account security settings <https://github.com/settings/security>`_.
+2FA means that you'll have to enter an authentication code when logging into GitHub.com from a new computer.
+Apps like `1Password <https://agilebits.com/onepassword>`_ (see their `guide <https://guides.agilebits.com/1password-ios/5/en/topic/setting-up-one-time-passwords>`_), `Authy <https://www.authy.com>`_, and the Google Authenticator App can help you generate these authentication codes.
+When pushing code with a 2FA-enabled account, you'll use a personal access token instead of your password.
+You can `create and revoke access easily from your GitHub settings page <https://github.com/settings/tokens>`_.
+To help you automatically authenticate when pushing to GitHub, we encourage you to follow the next step and enable a credential helper.
+
+.. _git-credential-helper:
+
+Setup a Git Credential Helper
+-----------------------------
+
+Rather than entering your GitHub username and password (or 2FA access token) every time you push, you can setup a Git credential helper to manage this for you.
+A credential helper is especially important for working with our :doc:`Git LFS-backed repositories </development/git_lfs>`.
+
+**Mac users** can use the secure OS X keychain:
+
+.. code-block:: bash
+
+   git config --global credential.helper osxkeychain  
+
+**Linux users** can use a credential *cache* to temporarily keep your credentials in memory.
+To have your credentials cached for 1 hour (3600 seconds):
+
+.. code-block:: bash
+
+   git config --global credential.helper 'cache --timeout=3600'
+
+**Linux users can alternatively** have their `credentials stored on disk <http://git-scm.com/docs/git-credential-store>`_ in a :file:`~/.git-credentials` file.
+Obviously, only do this for machines where you can ensure some level of security.
+
+.. code-block:: bash
+
+   git config credential.helper store
+
+Once a credential helper is enabled, the next time you ``git push``, you will add your credentials to the helper.
+
+Remember that if you have 2FA enabled, you will create and use a `personal access token <https://github.com/settings/tokens>`_ instead of your GitHub password.
+
+The DM Git LFS documentation has further information about :ref:`authenticating with our LFS storage backend <git-lfs-auth>`.
+
+.. _git-shell-setup:
+
+Tune your shell for Git
+-----------------------
+
+You can build an effective development environment and workflow by tuning your Git setup.
+Here are some ideas:
+
+1. `Add git status to your prompt <http://git-scm.com/book/en/v2/Git-in-Other-Environments-Git-in-Bash>`_.
+2. `Enable shell autocompletion <http://git-scm.com/book/en/v2/Git-in-Other-Environments-Git-in-Bash>`_
+3. `Craft aliases for common workflows <http://git-scm.com/book/en/v2/Git-Basics-Git-Aliases>`_.
```

Comment:
I've got a bunch of aliases I find helpful for our workflow.  Not sure you want to include them in this document as I'm sure there are matters of personal preference, but on the off chance it might be useful, here they are:

        stat = status
        co = checkout
        t = "tag -as"
        cp = cherry-pick
        desc = describe --tags --always
        recent = "log --graph --date-order --pretty=format:'%C(yellow)%h%d%Creset %Cred%an%Creset %Cblue%cr%Creset %s'"
        hist = "log --decorate --graph --date-order --stat"
        fixup = commit --fixup=HEAD
        # Amend last commit without modifying commit message
        amend = "!git log -n 1 --pretty=tformat:%s%n%n%b | git commit -F - --amend"
        # Outgoing commits
        out = "log @{u}.."
        # Undo merge
        unmerge = "reset --hard ORIG_HEAD"
        # Toss new commits
        toss = "reset --hard @{u}"
        # Clean up files that don't belong and toss any new commits
        nuke = "!git clean -df && git reset --hard @{u}"
        # Show new commits created by last command (e.g., after "git pull")
        new = !sh -c 'git log $1@{1}..$1@{0} "$@"'
        # List aliases
        alias = "config --get-regexp '^alias.*'"
        # Display for review
        review = !sh -c 'git log --patch --reverse "${2:-origin/master}".."${1:-HEAD}"' -
        # Display for submission
        sub = !sh -c 'git --no-pager log --stat --reverse "${2:-origin/master}".."${1:-HEAD}"' -
        sub-patch = !sh -c 'git --no-pager log --patch --reverse ${2:-origin/master}..${1:-HEAD}' -